crates/ruff/src/rules/pylint/rules/assert_on_string_literal.rs

```diff
@@ -0,0 +1,43 @@
+use rustpython_parser::ast::{Constant, Expr, ExprKind};
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::types::Range;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for `assert` statements that use a string literal as the first
+/// argument.
+///
+/// ## Why is this bad?
+/// An `assert` on a string literal will always pass.
+///
+/// ## Example
+/// ```python
+/// assert "always true"
+/// ```
+#[violation]
+pub struct AssertOnStringLiteral;
+
+impl Violation for AssertOnStringLiteral {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Asserting on a string literal will always pass")
```

Comment:
It looks like we need to change the `never` if the string is empty -- try running `pylint` over this snippet:

```py
assert "foo"
assert ""
```

Reply:
Ah right, `""` is falsey isn't it, so that would evaluate to `assert False` instead of `assert True`

Reply:
Yeah, that's right!

Reply:
"Asserting on a string literal may result in unintended results" Something like that?

Reply:
I think it can be `Asserting on a non-empty string literal will always pass` and `Asserting on an empty string literal will never pass`